mesh/mesh.go

```diff
@@ -117,13 +113,18 @@ func NewMesh(db *DB, atxDb AtxDB, rewardConfig Config, trtl tortoise, txPool txM
 		DB:                  db,
 		config:              rewardConfig,
 		AtxDB:               atxDb,
-		nextProcessedLayers: make(map[types.LayerID]*types.Layer),
+		nextProcessedLayers: make(map[types.LayerID]struct{}),
 		nextValidLayers:     make(map[types.LayerID]*types.Layer),
 		latestLayer:         types.GetEffectiveGenesis(),
 		latestLayerInState:  types.GetEffectiveGenesis(),
 	}
 
 	msh.Validator = &validator{Mesh: msh}
+	gLyr := types.GetEffectiveGenesis()
+	for i := types.NewLayerID(1); i.Before(gLyr); i = i.Add(1) {
+		msh.SetZeroBlockLayer(i)
+	}
+	msh.persistLayerHashes(context.Background(), types.NewLayerID(1), gLyr)
```

Comment:
strange that it doesn't return error

Reply:
how do you mean? do you expect errors here?

Reply:
we are writing to the database, if smth is wrong with disk errors are guaranteed, no?

Reply:
yes. ok. you mean generic error handling. i thought you saw something specific. thanks